### ISSUE TITLE
When checking that a Gazetteer place is ok to be deleted we need to add a check against the birth places in search records too

### ISSUE THREAD

**Issue by AnneV-Learn** [FreeUKGen/FreeCENMigration#1481]

**Comment by AnneV-Learn** [comment]
Currently only a check against search records with a matching 'census place' is done and the birth_place is not considered at all.

**Comment by DeniseColbert** [comment]
To be discussed re. Place of Birth Searching with the FreeCEN Committee

**Comment by AnneV-Learn** [comment]
@geoffj-FUG I think we diid to resolve this before we go any further with the VLD POB Validation.
Currently when a user (restricted by role) attempts to delete a Gazetteer place it checks to see if the County Place is used in Search records. The check it does only checks against the Census County Place in Search records. But Search records also hold the _County and Place of Birth_ (as values not as links). So currently it is possible to delete a Gazetteer place even if it is used as a POB.  
Do you agree that we need prohibit the deletion of Gazetteer places that are used as POB in Search records?

**Comment by geoffj-FUG** [comment]
Anne

 

Most definitely.

 

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG Prohibit the deletion of Gazetteer places that are used as POB in Search records - this is ready for testing on Test3 now.

**Comment by DeniseColbert** [comment]
@geoffj-FUG to test

**Comment by geoffj-FUG** [comment]
Anne

I have tried to delete Bristol using the Bristol name button and Delete entry from that menu. I was stopped by the entry having search records. This has been the case for some time now so no change evident by the test.
I then tried to destroy Bristol Bedminster which is an entry attached to Bedminster. There will be several search entries against that name but they will be attached to the main Bedminster entry. The entry was destroyed.

It looks as if we need to attach the test o the Other Place names in the Gazetteer as well.

Geoff

**Comment by AnneV-Learn** [comment]
@GeoffJ-FUG You are right (of course). I’ll make the appropriate changes and let you know when it is ready for re-testing.

**Comment by AnneV-Learn** [comment]
Ok @geoffj-FUG - Ready for re-Testing on Test3.
FYI - If the deletion/destroy is prevented because there are search records with that Place or Alternative Place name recorded in as a birth place (in the same county of course) the error message states that, eg
The Other Name for Place (xxxxx) cannot be deleted because there are dependent search record **birth places**
or
The Place cannot be deleted because there are dependent search record **birth places**

**Comment by geoffj-FUG** [comment]
Anne

 

Can we change Destroy Alternative place name back to Edit please/ Destroy is a form of Edit but so is fixing a spelling.

 

When I try to destroy an alternative place name I get the following message after about 2 minutes (a guess re time).

Proxy Error

The proxy server received an invalid response from an upstream server.
The proxy server could not handle the request

Reason: Error reading from remote server

Geoff

**Comment by AnneV-Learn** [comment]
@geoffj-FUG 
I could not replicate but I have added a 'no timeout' to the look up for the Search records when Destroying an Alternative Place. Potentially lots of records to look through but this will be an action that is not used very frequently I guess and when we do add indexes to the search records for Birth place it will improve the response.
 Destroy Alternative place name has now been changed to Edit as requested (this was just a quirk on Test3).
FYI - Below is the Test that I did and it worked:
![Screenshot 2023-07-12 at 12.28.36.png](https://images.zenhubusercontent.com/6009aeccd60429cb7008e5d3/c2b57885-bdde-4fee-b32f-9fc8d09a5ace)
